src/main/fc/cli.c

```diff
@@ -3009,48 +3012,259 @@ static void cliResource(char *cmdline)
+
+        const timerTag_t timerTag = timerTags[i];
+        const uint8_t timer = TIMER_TAG_INDEX(timerTag);
+        const uint8_t channel = TIMER_TAG_CHANNEL_INDEX(timerTag);
+        const bool inverted = timerChannelConfig(i)->inverted;
+        const ioTag_t ioTag = timerChannelConfig(i)->ioTag;
+
+        if (!ioTag) {
+            if (!(dumpMask & HIDE_UNUSED)) {
+                cliDumpPrintLinef(dumpMask, false, formatUnassigned, timer, channel);
+            }
+        } else {
+            memset(buffer, 0, sizeof(buffer));
+            if (timerChannelConfig(i)->dma) {
+                tfp_sprintf(buffer,
+                    dmaFormat, 
+                    DMA_DEVICE(timerChannelConfig(i)->dma),
+                    DMA_DEVICE_INDEX(timerChannelConfig(i)->dma)
+#if defined(STM32F7) || defined(STM32F4)
+                    , timerChannelConfig(i)->dmaChannel
+#endif
+                    );
+            } else {
+                tfp_sprintf(buffer, "%s", "NONE");
+            }
+            
+            cliDumpPrintLinef(dumpMask, false, format, 
+                timer, 
+                channel,
+                (inverted) ? "N" : "",
+                IO_GPIOPortIdxByTag(ioTag) + 'A', IO_GPIOPinIdxByTag(ioTag),
+                timerChannelConfig(i)->alternateFunction,
+                buffer
+                );
+        }
+    }
+}
+
+static void cliTimer(char *cmdline)
+{
+    int len = strlen(cmdline);
+
+    if (len == 0) {
+        printTimer(DUMP_MASTER | HIDE_UNUSED);
+        return;
+    } else if (strncasecmp(cmdline, "list", len) == 0) {
+        printTimer(DUMP_MASTER);
+        return;
+    }
+    
+    char *pch = NULL;
+    char *saveptr;
+    char buffer[20];
+    
+    memset(buffer, 0, sizeof(buffer));
+    timerTag_t timerTag = 0;
+    
+    const char *timChformat = "tim%d_ch%d%s";
+    bool inverted = false;
+    int timerChannelIndex = 0;
+    
+    pch = strtok_r(cmdline, " ", &saveptr);
+    for (int i = 0; i < TIMER_CHANNEL_COUNT; i++) {
+        const uint8_t timer = TIMER_TAG_INDEX(timerTags[i]);
+        const uint8_t channel = TIMER_TAG_CHANNEL_INDEX(timerTags[i]);    
+
+        tfp_sprintf(buffer, timChformat, timer, channel, "");
+        if (strncasecmp(pch, buffer, len) == 0) {
+            timerTag = timerTags[i];
+            timerChannelIndex = i;
+            break;
+        }
+            
+        tfp_sprintf(buffer, timChformat, timer, channel, "N");
+        if (strncasecmp(pch, buffer, len) == 0) {
+            timerTag = timerTags[i];
+            timerChannelIndex = i;
+            inverted = true;
+            break;
+        }
+    }
+    
+    if (!timerTag) {
+        goto error;
+    }    
+    
+    ioTag_t ioTag = 0;
+    pch = strtok_r(NULL, " ", &saveptr);
+    if (strlen(pch) > 0) {
+        if (strToPin(pch, &ioTag)) {
+            if (ioTag != IO_TAG_NONE) {
+                ioRec_t *rec = IO_Rec(IOGetByTag(ioTag));
```

Comment:
Yes agreed thx - there's plenty of areas to optimise.

In most of this (not in this case) I've opted for readability so it's easier to see the thinking and direction, for discussion. Can clean up this and others once optimising the implementation.